mistralrs-core/src/request.rs

```diff
@@ -13,11 +13,15 @@ use crate::{
 use std::{fmt::Debug, sync::Arc};
 use tokio::sync::mpsc::Sender;
 
+pub type LlguidanceGrammar = llguidance::api::TopLevelGrammar;
+
 #[derive(Clone)]
-/// Control the constraint with Regex or Yacc.
+/// Control the constraint with llguidance.
 pub enum Constraint {
     Regex(String),
-    Yacc(String),
+    Lark(String),
+    JsonSchema(serde_json::Value),
+    Llguidance(LlguidanceGrammar),
```

Comment:
Can we export the `llguidance` and (if necessary) `toktrie_hf_tokenizers` crates from`mistralrs-core`? I think this will help users constructing a `LlguidanceGrammar`.

Reply:
Only `llguidance` I think.